communication/src/main/java/org/jboss/da/communication/pom/impl/DependencyTreeBuilder.java

```diff
@@ -108,7 +110,7 @@ private void walkDependencies(PathNode root, Map<GAV, Set<DependencyRelationship
 
         private final GAV gav;
 
-        private final Set<PathNode> childs = new HashSet<>();
+        private final Set<PathNode> childs = new TreeSet<>(new PathNodeComparator());
```

Comment:
Didn't you want to try lambdas? Or you can change PathNode to implement Comparable. Or you can leave it as is. :)